docs/element-definitions.md

```diff
@@ -1353,6 +1356,339 @@ the `inferred` link tells the user that the element was created based on some
+
+### Basic Authentication Scheme (Base API Element)
+
+This element may be used to define a basic authentication scheme implementation for an API as described in [RFC 2617](https://tools.ietf.org/html/rfc2617).
+
+The element MAY have a username and password defined as member elements within the scheme, but these are not required.
+
+- `username` (string, optional)
+- `password` (string, optional)
+
+#### Properties
+
+- `element`: Basic Authentication Scheme (string, fixed)
+- `content` (array[Member Element])
+
+#### Example
+
+This example shows a custom basic authentication scheme being defined as `Custom Basic Auth`. This scheme is then used on an HTTP transaction within a resource. Please note this example is incomplete for the sake of keeping it short.
+
+```json
+{
+  "element": "category",
+  "meta": {
+    "classes": ["api"]
+  },
+  "content": [
+    {
+      "element": "category",
+      "meta": {
+        "classes": ["authSchemes"]
+      },
+      "content": [
+        {
+          "element": "Basic Authentication Scheme",
+          "meta": {
+            "id": "Custom Basic Auth"
+          },
+          "content": [
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "username"
+                },
+                "value": {
+                  "element": "string",
+                  "content": "john.doe"
+                }
+              }
+            },
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "password"
+                },
+                "value": {
+                  "element": "string",
+                  "content": "1234password"
+                }
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "element": "resource",
+      "attributes": {
+        "href": "/users"
+      },
+      "content": [
+        {
+          "element": "transition",
+          "content": [
+            {
+              "element": "httpTransaction",
+              "attributes": {
+                "authSchemes": [
+                  {
+                    "element": "Custom Basic Auth"
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Token Authentication Scheme (Base API Element)
+
+This describes an authentication scheme that uses a token as a way to authenticate and authorize. The token MAY exist as an HTTP header field or a query parameter.
+
+- One of
+    - `httpHeaderName` (string)
+    - `queryParameterName` (string)
+
+When used as a query parameter, an HREF Variable is not required to be defined within the scope of the resource or transition, but is rather infered from the used token authentications scheme.
```

Comment:
I understand the reasoning behind not requiring the HREF variable and URI template variable to be defined, but I think that this is inconsistent with how the rest of the elements work. We should be sure we want to do this!

Also, is it still possible to define it? What if a user wants to put limits, validations, example values, etc? E.g. the auth token must be 12 alphanumeric characters. Just a thought.